flake.nix

```diff
@@ -1,15 +1,78 @@
 {
-  description = "r/unixporn discord bot";
 
+  description = "r/unixporn discord bot";
   inputs = {
-    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    crane = {
+      url = "github:ipetkov/crane";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.05";
     utils.url = "github:numtide/flake-utils";
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+    rust-overlay.url = "github:oxalica/rust-overlay";
   };
 
-  outputs = { self, nixpkgs, utils }: utils.lib.eachDefaultSystem (system:
-    let pkgs = nixpkgs.legacyPackages.${system}; in
-    {
-      devShell = import ./shell.nix { inherit pkgs; };
-    }
-  );
+  outputs = { self, nixpkgs, utils, crane, rust-overlay, ... }:
+    utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = [ (import rust-overlay) ];
+        };
+
+        # Common Args
+        commonArgs = {
+          pname = "robbb";
+          # Will Require `--impure` to be passed to it can read that ENV VAR
+          version = if builtins.getEnv "VERSION" != "" then
```

Comment:
might as well set it to some arbitrary version, instead of reading from env

Reply:
not sure about this myself. I did set a fallback to "0.0.1"

Reply:
`self.rev` is probably the best move

Reply:
So just remove the impure part and use `self.rev`?

Reply:
yes